### ISSUE TITLE
Advice implementing a drain plugin?

### ISSUE THREAD

**Issue by chadacious** [elysiajs/elysia-apollo#10]
How would we implement a drain plugin before shutdown? Of course, there is this plugin for apollo and express: https://www.apollographql.com/docs/apollo-server/api/plugin/drain-http-server/

But how would we do it in elysiajs?

I can write a simple drain plugin for apollo, but the problem seems that this elysia plugin never calls the stop() function of the apollo server on the stop signal from elysia.

**Comment by chadacious** [comment]
I'm testing with a simple change:
```

            app.onStop(() => this.stop())

            return app.post(
                path,
```
Will do a PR once I experiment a little, but this seems to be what we need for this to work properly.

**Comment by chadacious** [comment]
Well, I ended up going in a little different direction once I realized what I was really looking for here.

My main use case was when deploying new images to my k8s cluster, the pods all received the termination signal from the load balancer and would start exiting. But the problem was that any requests they were processing at that time were getting dropped, which was not good.

So I built an elysia/apollo gql request drain plugin and tried it out a couple times today and its working perfectly in my case! No changes to this package was needed.

It even detects when there is a readiness probe. So when the cluster sends the termination signal, the server will keep running until the next readiness probe, at which point it will return a 503 to indicate it is no longer accepting requests. Then it will wait for all the pending requests to drain before proceeding with the shutdown.

Here is a [gist](https://gist.github.com/chadacious/7c9bc5c4b376bd9a07df17ee70539fec.js) with the plugin as a work in progress:
